inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -1024,25 +1040,33 @@ bool pass::Serialize::run_on_function(std::shared_ptr<ngraph::Function> f) {
     RUN_ON_FUNCTION_SCOPE(Serialize);
 
     auto serializeFunc = [&] (std::ostream & xml_file, std::ostream & bin_file) {
-        switch (m_version) {
-        case Version::IR_V10:
-            {
-                std::string name = "net";
-                pugi::xml_document xml_doc;
-                pugi::xml_node net_node = xml_doc.append_child(name.c_str());
-                ConstantWriter constant_write_handler(bin_file);
-                XmlSerializer visitor(net_node, name, m_custom_opsets, constant_write_handler);
-                visitor.on_attribute(name, f);
-
-                xml_doc.save(xml_file);
-                xml_file.flush();
-                bin_file.flush();
-            }
-            break;
-        default:
-            NGRAPH_UNREACHABLE("Unsupported version");
-            break;
+        auto version = static_cast<int64_t>(m_version);
+        auto& rt_info = f->get_rt_info();
+        if (rt_info.count("version")) {
+            auto version_var = std::dynamic_pointer_cast<VariantWrapper<int64_t>>(rt_info.at("version"));
+            version = version_var->get();
+        }
+
+        if (version != static_cast<int64_t>(m_version) && m_version != Serialize::Version::UNSPECIFIED)
+            throw ngraph_error("Cannot serialize function to incompatible IR version");
```

Comment:
can we extend the error like `you are trying to serialize IR v10 as IR 11` or smth like this?